spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb

```diff
@@ -337,5 +379,65 @@ class { '::apache':
         )
       end
     end
+
+    context 'with heredoc' do
+      context 'with trailing comma' do
+        let(:code) {
+          <<-EOS
+          file { '/tmp/test.txt':
+            ensure  => 'file',
+            content => @(EOT),
+              Hello
+              World
+              EOT
+          }
+          EOS
+        }
+
+        it 'should not detect any problems' do
+          expect(problems).to have(0).problems
+        end
+
+        it 'should not modify the manifest' do
+          expect(manifest).to eq(code)
+        end
+      end
+
+      context 'without trailing comma' do
+        let(:code) {
+          <<-EOS
+          file { '/tmp/test.txt':
+            ensure  => 'file',
+            content => @(EOT)
+              Hello
+              World
+            EOT
+          }
+          EOS
+        }
+
+        it 'should detect a problem' do
+          expect(problems).to have(1).problems
+        end
+
+        it 'should create a warning' do
+          expect(problems).to contain_fixed(msg).on_line(3).in_column(30)
+        end
+
+        it 'should add trailing commas' do
+          expect(manifest).to eq(
+            <<-EOS
+          file { '/tmp/test.txt':
```

Comment:
A bit odd indenting here

Reply:
Yes, it is. But it isn't trivially fixable unless we can use the `~EOS` heredoc format, but I thought you still supported Ruby versions that don't support that. See the prior art here: https://github.com/voxpupuli/puppet-lint-trailing_comma-check/blob/master/spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb#L288

The problem is that the spacing in the expectation has to match the spacing in the original `let(:code)` block because it is doing an exact string comparison.